front/pages/[user]/ds/[name]/search.tsx

```diff
@@ -0,0 +1,303 @@
+import AppLayout from "@app/components/AppLayout";
+import MainTab from "@app/components/data_source/MainTab";
+import { auth_user } from "@app/lib/auth";
+import { timeAgoFrom, utcDateFrom } from "@app/lib/utils";
+import { unstable_getServerSession } from "next-auth/next";
+import { authOptions } from "@app/pages/api/auth/[...nextauth]";
+import Link from "next/link";
+import { useSession } from "next-auth/react";
+import { useSWRConfig } from "swr";
+import { useState, useEffect } from "react";
+
+import { DocumentType } from "@app/types/document";
+import { DataSourceType } from "@app/types/data_source";
+import { classNames } from "@app/lib/utils";
+import { UserType } from "@app/types//user";
+
+const { URL, GA_TRACKING_ID = null } = process.env;
+
+export default function DataSourceView({
+  dataSource,
+  readOnly,
+  authUser,
+  owner,
+  gaTrackingId: gaTrackingId,
+}: {
+  dataSource: DataSourceType;
+  readOnly: boolean;
+  authUser?: UserType;
+  owner: {
+    username: string;
+  };
+  gaTrackingId: string;
+}) {
+  const { mutate } = useSWRConfig();
+  const { data: session } = useSession();
+
+  const [searchQuery, setSearchQuery] = useState("");
+  const [documents, setDocuments] = useState<DocumentType[]>([]);
+  const [expandedChunkId, setExpandedChunkId] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
+
+  useEffect(() => {
+    setError(false);
+    let isCancelled = false;
+    (async () => {
+      if (searchQuery.trim().length == 0) {
+        setDocuments([]);
+
+        return;
+      }
+      setIsLoading(true);
+      const searchParams = new URLSearchParams();
+      searchParams.append("query", searchQuery);
+      searchParams.append("top_k", "10");
+      searchParams.append("full_text", "false");
+
+      const searchRes = await fetch(
+        // @todo(aric): Not sure how to properly construct the URL here.
+        // I don't know if we want to let users search someeone else's data source.
+        // And I don't know what to do about anonymous users.
+        // Please advise @spolu.
+        // @ts-expect-error session.user.username is added dynamically by us in the auth callback.
+        `/api/data_sources/${session.user.username}/${dataSource.name}/search?` +
+          searchParams.toString(),
+        {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        }
+      );
+      setIsLoading(false);
+      if (isCancelled) {
+        return;
+      }
+
+      if (searchRes.ok) {
+        setError(false);
+        const documents = await searchRes.json();
+        setDocuments(documents.documents);
+      } else {
+        setError(true);
```

Comment:
I think we want to show errors properly if we allow public dataSources to be searched by other users, as we don't have guarantee that the user will have the necessary credentials to run the search (no credentials to embed the query).

Reply:
Is the backend returning the proper error message to let the user know that their credentials are not defined?